### DISCUSSION TITLE
Running npm install in GitHub codespaces with a repository that uses GitHub packages

### DISCUSSION THREAD

**Discussion by dlemstra** [community/community#34043] [Codespaces]
I am not sure if I should report this as an issue for [Oryx](https://github.com/microsoft/Oryx/) or [vscode-dev-containers](https://github.com/microsoft/vscode-dev-containers/) so that is why I am starting the discussion here. When a Codespace that uses node is started the Oryx tool will do the following:

```
2022-09-24 07:03:11.098Z: 
2022-09-24 07:03:11.100Z: Using Npm version:
2022-09-24 07:03:11.259Z: 6.14.16
2022-09-24 07:03:11.266Z: 
2022-09-24 07:03:11.270Z: Running 'npm install --unsafe-perm'...

... snip ...

2022-09-24 07:03:17.043Z: npm ERR! code E401
2022-09-24 07:03:17.047Z: npm ERR! 401 Unauthorized - GET https://npm.pkg.github.com/download/@dlemstra/magick-native/0.202207.41848/b8504059d932bade870b440fb2acaa19628009015746dce5f88a656920733809
2022-09-24 07:03:17.085Z: 
```

This is happening because this [project](https://github.com/dlemstra/magick-wasm) is using GitHub Packages for one of its dependencies. And because GitHub packages requires an auth token the installation will fail. I solved this by creating a `.npmrc` file in the project with the following content:

```
//npm.pkg.github.com/:_authToken=${GITHUB_TOKEN}
```

It would be nice if this (or a better solution) would be done automatically by GitHub codespaces.

**Comment by jungaretti** [comment]
I'd suggest you create a `devcontainer.json` file in your repo with a `postCreateCommand` that initializes your `.npmrc` file and then executes `npm install`. Here's an example:

```json
{
  "postCreateCommand": "echo \"//npm.pkg.github.com/:_authToken=${GITHUB_TOKEN}\" > ~/.npmrc && npm install"
}
```

Oryx only executes if your `devcontainer.json` file does not exist or does not specify a `postCreateCommand`.

Useful links:
- https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers
- https://containers.dev/implementors/json_reference/

**Reply by dlemstra** [reply]
I already have a workaround that works for me. I was just pointing out that it would make sense that this would work by default because they are both GitHub products.
